### DISCUSSION TITLE
Tandem Repeat Benchmark Variant Representation

### DISCUSSION THREAD

**Discussion by TimD1** [ACEnglish/adotto#4] [General]
I'm trying to figure out which Smith-Waterman parameters were used to select the representation of variants in the tandem repeat ground truth VCF.

From my understanding, [this line](https://github.com/ACEnglish/adotto/blob/f2fcdb02df66e28690d9d81f53097aee3fec84ba/variants/scripts/map_haplo.sh#L34) in `variants/scripts/map_haplo.sh` uses `minimap2` to align each haplotype to the reference FASTA, therefore selecting the variant representation.

My confusion arises from the fact that `params='-x asm20 ... -O 5,56 -E 4,1 -B'` in [this line](https://github.com/ACEnglish/adotto/blob/f2fcdb02df66e28690d9d81f53097aee3fec84ba/variants/scripts/map_haplo.sh#L13). Based on the `minimap2` documentation, if you want to override the mismatch penalty, you must provide `-B INT`. For me, not providing this value results in `[ERROR] missing option argument`. Are you overriding the mismatch penalty? If so, with what value? If not, then what does the `-B` flag do here?

Thanks in advance for clarifying!

**Comment by ACEnglish** [comment]
The value of `-B 5` was supposed to there. Looks like the line in the code got erroneously truncated. Minimap2.24 was used for the initial alignments and it does not throw an error with only `-B`. Looking at the logs from the benchmark's alignment I believe the truncated line was used so presumably `-B 4` (minimap's default) was used.

**Reply by TimD1** [reply]
Thanks for the quick response!
